src/types/gecko-profile.js

```diff
@@ -224,16 +217,46 @@ export type GeckoProfileMeta = {|
     | 'release'
     | 'esr' // Extended Support Release channel
     | string,
+  // -- platform information -- This can be absent in some very rare situations.
+  platform?: string,
+  oscpu?: string,
+  misc?: string,
+  // -- Runtime -- This can be absent in some very rare situations.
+  abi?: string,
+  toolkit?: string,
+  product?: string,
+  // -- appInfo -- This can be absent in some very rare situations.
+  // The appBuildID, sourceURL, physicalCPUs and logicalCPUs properties landed
+  // in Firefox 62, and are only optional because older processed profile
+  // versions may not have them. No upgrader was written for this change.
+  appBuildID?: string,
+  sourceURL?: string,
+  // -- system info -- This can be absent in some very rare situations.
+  physicalCPUs?: number,
+  logicalCPUs?: number,
+  // -- extensions --
+  // The extensions property landed in Firefox 60, and is only optional because
+  // older profile versions may not have it. No upgrader was written for this change.
+  extensions?: GeckoExtensionMeta,
+  // -- extra properties added by the frontend --
+  // This boolean indicates whether this gecko profile includes already
+  // symbolicated frames. This will be missing for profiles coming from Gecko
+  // (which indicates that they'll need to be symbolicated) but may be specified
+  // for profiles imported from other formats (eg: linux perf).
+  presymbolicated?: boolean,
 |};
 
-export type GeckoProfile = {|
+export type GeckoGenericProfile<Meta> = {|
```

Comment:
What about `GeckoProfileWithMeta<Meta>`? I think it would be a bit more precise than the word "generic".  It might be nice to have a short code comment explaining that subprocesses have slightly different meta information.